src/clr-angular/forms/select/providers/option-selection.service.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable} from "@angular/core";
+import {Observable, Subject} from "rxjs";
+
+import {ClrOption} from "../option";
+
+@Injectable()
+export class OptionSelectionService<T> {
+    private currentValue: T;
+    private currentOption: ClrOption<T>;
```

Comment:
Do we need to track `currentOption` here? `currentOption` is based on `currentValue`, right? When `currentValue` changes current option should change...?

Reply:
So I believe we assume that the option itself is static. The option list can change but once option content wouldn't. Thats why we do not need to track this info and thats why we directly clone the option on the input.